app/routes/users/components/PhotoConsents.js

```diff
@@ -0,0 +1,173 @@
+import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
+import type { PhotoConsent } from 'app/models';
+import moment from 'moment-timezone';
+import cx from 'classnames';
+import {
+  getConsent,
+  PHOTO_CONSENT_DOMAINS,
+  toReadableSemester,
+} from 'app/routes/events/utils';
+import styles from './PhotoConsents.css';
+
+const ConsentManager = ({
+  consent,
+  updateConsent,
+  isMe,
+}: {
+  consent: ?PhotoConsent,
+  updateConsent: (consent: PhotoConsent) => Promise<*>,
+  isMe: boolean,
+}) => {
+  if (!consent) {
+    return null;
+  }
+
+  const consentStatus =
+    typeof consent.isConsenting === 'boolean'
+      ? consent.isConsenting
+        ? 'Du ga samtykket den '
+        : 'Du trakk samtykket den '
+      : 'Du har ikke tatt stilling til samtykket.';
+
+  const presentableDomain =
+    consent.domain === PHOTO_CONSENT_DOMAINS.WEBSITE
+      ? 'Abakus.no'
+      : 'Sosiale medier';
+  return (
+    <>
+      <h4 className={styles.categoryTitle}>{presentableDomain}</h4>
+      <h5>
+        Jeg godtar at Abakus kan legge ut bilder av meg på {presentableDomain} i
+        perioden {toReadableSemester(consent)}:
+      </h5>
+      <div className={styles.statusContainer}>
+        {consentStatus}
+        {typeof consent.isConsenting === 'boolean' ? (
+          <i>{moment(consent.updatedAt).format('DD. MMM YYYY')}.</i>
+        ) : (
+          ''
+        )}
+      </div>
+      <div className={styles.consentBtnContainer}>
+        <ConfirmModalWithParent
+          closeOnConfirm={true}
+          title={`Trekke bildesamtykke på ${presentableDomain}`}
+          message={`Er du sikker på at du vil trekke bildesamtykket ditt for ${toReadableSemester(
+            consent
+          )} på ${presentableDomain}? Dersom du ønsker å fjerne noen spesifikke bilder, kan du i stedet sende en mail til pr@abakus.no med informasjon om hvilke bilder du vil fjerne.`}
+          onConfirm={() => updateConsent({ ...consent, isConsenting: false })}
+        >
+          <Button
+            dark
+            disabled={!isMe}
+            className={
+              consent.isConsenting === false
+                ? cx(styles.consentBtn, styles.selectedConsentBtn)
+                : styles.consentBtn
+            }
+          >
+            Trekk samtykke
+          </Button>
+        </ConfirmModalWithParent>
+        <Button
+          success
+          disabled={!isMe}
+          onClick={() => updateConsent({ ...consent, isConsenting: true })}
+          className={
+            consent.isConsenting
+              ? cx(styles.consentBtn, styles.selectedConsentBtn)
+              : styles.consentBtn
+          }
+        >
+          Behold samtykke
+        </Button>
+      </div>
+    </>
+  );
+};
+
+const PhotoConsents = ({
+  photoConsents,
+  username,
+  updatePhotoConsent,
+  userId,
+  isMe,
+}: {
+  photoConsents: Array<PhotoConsent>,
+  username: string,
+  updatePhotoConsent: (
+    PhotoConsent: PhotoConsent,
```

Comment:
I'm guessing this should be `photoConsent`?

Reply:
we good